cli/azd/internal/appdetect/docker.go

```diff
@@ -70,3 +82,62 @@ func parsePortsInLine(s string) ([]Port, error) {
 	}
 	return ports, nil
 }
+
+const (
+	DockerfileSingleStage = `FROM openjdk:17-jdk-slim
+COPY ./target/*.jar app.jar
```

Comment:
what if users application requires a higher JDK version?

Reply:
I intend to determine it by the `java.version` property, but it's not mandatory in spring boot project. Maybe we hardcode a higher version? or do you have any suggestion?

Reply:
My suggestion:
1. Detect java version by these ways: 
  1.1. From pom files.
  1.2. From spring-boot version
  1.3. From "java" command in Path.
  1.4. Maybe other ways...
2. If can not be detected, use default version. (If Java version is back-ward compatible, use the higher version is better. For example: Java 21).

Reply:
Here I will first update the base image to jdk-21. Later, we will use another item to track the Java version detection.
https://devdiv.visualstudio.com/DevDiv/_workitems/edit/2330461